requires-all.txt

```diff
@@ -9,6 +9,7 @@ flake8==3.9.2
 flaky==3.7.0
 flask-talisman==0.8.1
 isort==4.3.21;python_version<"3.7"
+importlib-metadata
```

Comment:
This file is not in the actual requirements and may be outdated, it is generated by CI to create a hash for requirements. If this is a new requirements it should go in `requires-install.txt` or the other `requires-*` for extras.

Reply:
What @T4rk1n is saying is this whole file is useless, [CI creates it](https://github.com/plotly/dash/blob/d9b8d17e3fae5c7f9f342e480a85e782f6289fe8/.circleci/config.yml#L45) but it shouldn't really be in the repo. Shall we just delete it and `.gitignore` it?

That said there's a weird situation with this package: we pin it for Py3.6:

https://github.com/plotly/dash/blob/d9b8d17e3fae5c7f9f342e480a85e782f6289fe8/requires-install.txt#L7

but we don't require it directly for later Python versions, perhaps we should add the line:
```
importlib-metadata;python_version>="3.7"
```
to `requires_install.txt`?

Reply:
Updated 👍

Reply:
> perhaps we should add the line:
> ```
> importlib-metadata;python_version>="3.7"
> ```

But it's backport for older versions of python. I believe with the latest version of python there's no requirement of `importlib-metadata` at all:
 
* Maybe we should remove that line?
* Or maybe we should remove `importlib-metadata` at all because seems it useful only for deprecated versions of python.